### ISSUE TITLE
Podman: fail to build images for other platforms

### ISSUE THREAD

**Issue by DCsunset** [NixOS/nixpkgs#222830]
### Describe the bug
Podman supports building packages for other platforms through build command like:

```sh
podman build --platform linux/amd64,linux/arm64 --manifest image-name .
```

Running this command will result in the following error (suppose sh used in the Dockerfile):

```
exec container process `/bin/sh`: Exec format error
```

This is because building images for another platform requires `qemu-user-static` package as said in the [doc](https://docs.podman.io/en/latest/markdown/podman-build.1.html#platform-os-arch-variant), which can't be found in nixpkgs.

### Steps To Reproduce
1. Create a Dockerfile to run some command like (`apk add nodejs`) using alpine image
2. Run `podman build --platform linux/amd64,linux/arm64 --manifest image-name  .` to build it

### Expected behavior
It should build images for multiple platforms successfully.

### Notify maintainers

@zowoq @pennae 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.19, NixOS, 23.05 (Stoat), 23.05pre463258.7067edc68c0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos, nixos-unstable"`
 - channels(luna): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by zowoq** [comment]
Looks like https://github.com/NixOS/nixpkgs/pull/160802 would address this?

**Comment by DCsunset** [comment]
Yes it seems so. I think the podman module can provide an option to enable qemu-user-static after that PR is merged.

**Comment by jennydaman** [comment]
https://github.com/NixOS/nixpkgs/pull/334859 supposedly got this working with `docker`.

On 2025-02-22 "cross-architcture containers using binfmt/qemu" has been added to wiki.nixos.org: https://wiki.nixos.org/wiki/Podman#Cross-architecture_containers_using_binfmt/qemu

Today on 2026-05-06, I tried the instructions and they work (note that reboot is required, even if you do `nixos-rebuild switch`).
